core/iwasm/common/wasm_c_api.c

```diff
@@ -511,7 +591,18 @@ wasm_store_delete(wasm_store_t *store)
     }
 
     wasm_runtime_free(store);
-    wasm_runtime_destroy_thread_env();
+
+#ifdef os_thread_local_attribute
+    if (--thread_local_stores_num == 0)
```

Comment:
Should we make sure that thread_local_stores_num is larger than 0? Will there be case that thread_local_stores_num is 0 and then we decrease it again?
How about changing to
`if (thread_local_stores_num > 0 && --thread_local_stores_num == 0)`?

Reply:
it is an unsigned integer.  `static os_thread_local_attribute unsigned thread_local_stores_num = 0;`

Reply:
Yes, I mean if it is decrease to 0, and then wasm_runtime_destroy_thread_env is called, and after that, will it be decreased again? If yes, then it will be decreased to UINT32_MAX, and if wasm_store_new is called again, the wasm_runtime_init_thread_env won't be called since it isn't 0 and error may occur when calling wasm func.

Reply:
Or had better add bh_assert(thread_local_stores_num > 0) before decrease it?